cf/post-register-lambda-v2.yaml

```diff
@@ -0,0 +1,229 @@
+    Default: development
+    AllowedValues:
+      - development
+      - staging
+      - production
+    Description: The environment for which the tables need to be created.
+
+Outputs:
+  PostRegistrationLambdaArnV2:
+    Description: ARN of PostRegistrationLambda
+    Value: !GetAtt PostRegistrationLambda.Arn
+    Export:
+      Name: !Sub PostRegistrationLambdaArnV2-${Environment}
+
+Conditions:
+  isProd: !Equals [!Ref Environment, "production"]
+
+Resources:
+  PostRegistrationLambdaRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: "sts:AssumeRole"
+      Path: /
+      Policies:
+        - PolicyName: !Sub "can-access-dynamo-lambda-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "dynamodb:DeleteItem"
+                  - "dynamodb:GetItem"
+                  - "dynamodb:PutItem"
+                  - "dynamodb:Scan"
+                  - "dynamodb:UpdateItem"
+                Resource:
+                  - Fn::Sub: "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/user-access-${Environment}*"
+                  - Fn::Sub: "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/invite-access-${Environment}*"
+        - PolicyName: !Sub "can-create-log-group-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogGroup"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*"
+        - PolicyName: !Sub "can-add-logs-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogStream"
+                  - "logs:PutLogEvents"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/post-register-invite-lambda-${Environment}:*"
+        - PolicyName: !Sub "can-post-to-post-registration-sns-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'sns:Publish'
+                Resource:
+                  - Fn::Sub: "arn:aws:sns:${AWS::Region}:${AWS::AccountId}:post-registration-user-access-${Environment}-default"
+
+  PostRegistrationTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      KmsMasterKeyId: "alias/aws/sns"
+      TopicName: !Sub "post-registration-user-access-${Environment}-default"
+
+  PostUserRegistrationSubscription:
+    Type: 'AWS::SNS::Subscription'
```

Comment:
What I read, the actual logic happens in the API and the lambda just hits an endpoint on every register

Reply:
Yes, this is a security measure. If we want lambda to hit the API endpoint directly, we'll need to put an auth middleware for the endpoint to guard it. That means Lambda will need a JWT token to pass that auth and we have to find a way to get that auth, otherwise we'd need a different way to authenticate the call, etc. Doing this via SNS removes the need from us to implement our own auth, and we can rely on AWS SNS message authentication instead, which we already have in place.